woocommerce/woocommerce
# Add menu item to publish button with 'Move to trash'

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #43047



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Make sure feature `product-pre-publish-modal` is enabled under `Features` tab from `/wp-admin/tools.php?page=woocommerce-admin-test-helper` (WooCommerce Beta Tester plugin) is required
3. Go to `Products` > `Add new` from the left side menu
4. Then add a `Name` to the product
5. From the dropdown menu right next to the `Publish` button a new menu item should be shown `Move to trash` 
<img width="575" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/88c2a54f-ec47-495c-8bf9-5e3009923607">

6. When click it a notice at the bottom left corner of the screen should be shown that reads `Product successfully deleted`
7. After a short time a redirection to the product list page should be shown. From there the trashed product should be listed in the trash tab of that page.
8. From the trash tab the product can be edited by clicking on its name. By doing so a redirection to the product editor page should happen.
9. Once there notice that the `Move to trash` item is not longer visible from the dropdown menu of the `Publish`/`Update` button. 
<img width="580" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/e8b16e3c-7105-4cef-8ce0-10119f426a34">




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
